plugins/discovery-ec2/src/test/java/org/elasticsearch/discovery/ec2/Ec2DiscoveryTests.java

```diff
@@ -144,6 +144,7 @@ protected List<TransportAddress> buildDynamicHosts(Settings nodeSettings, int no
                                 exchange.getResponseHeaders().set("Content-Type", "text/xml; charset=UTF-8");
                                 exchange.sendResponseHeaders(HttpStatus.SC_OK, responseBody.length);
                                 exchange.getResponseBody().write(responseBody);
+                                exchange.getResponseBody().flush();
```

Comment:
We can also consider closing `exchange` either explicitly or via `try-with-resources`. I believe that's what we do normally do in other `HttpHandlers`.

Reply:
The control-flow here is a bit weird, I'm ok with failing (and leaking the `exchange`) on exception.